src/home/current-positions/index.tsx

```diff
@@ -98,9 +84,18 @@ const CurrentPositions = ({ isLoading }: CurrentPositionsProps) => {
     return supportedNetwork;
   }, [currentNetwork.chainId]);
 
-  for (let i = 0; i < positionsToFill; i += 1) {
-    emptyPositions.push(i);
-  }
+  const onCancelModifySettingsModal = React.useCallback(
+    () => setShowModifyRateSettingsModal(false),
```

Comment:
This is a fun one, so basically react checks  by reference whenever it needs to re-update a component, since this functions where sent directly to the component, each time the parent component re-rendered for any value it would actually re-render anyone using this functions, so we were making unnecesary re-renders on the modals by not memoizing the cancel functions